### PR TITLE
Fix blog index style so it looks like it did before migrating away fr…

### PR DESCRIPTION
**Pull request by jwage**
Before

screen shot 2018-09-01 at 3 01 02 pm

After

screen shot 2018-09-01 at 3 01 22 pm

### CODE REVIEW

File: lib/Controllers/BlogController.php [hunk 1]
```diff
@@ -19,6 +20,13 @@ public function __construct(BlogPostRepository $blogPostRepository)
     }
 
     public function index(SourceFile $sourceFile) : ControllerResult
+    {
+        return new ControllerResult([
+            'blogPosts' => array_slice($this->blogPostRepository->findAll(), 0, 10),
```
Comment (SenseException) [inline lib/Controllers/BlogController.php]: We should put this slicing into the repository with an limit argument.

Comment (jwage) [reply lib/Controllers/BlogController.php]: Done.

File: templates/layouts/blog-post.html.twig [hunk 2]
```diff
@@ -43,7 +43,7 @@
         </header>
 
         <p class="lead">
-            Posted on {{ blogPost.date|date('Y-m-d') }}
+            Posted on {{ blogPost.date|date('F j, Y') }}
```
Comment (Ocramius) [inline templates/layouts/blog-post.html.twig]: Same here: need `<date/>` around it

File: source/archive.html [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+title: Blog Archive
+menuSlug: blog
+permalink: /blog/archive.html
+controller: ['Doctrine\Website\Controllers\BlogController', 'archive']
+---
+
+<h1>Blog Archive</h1>
+
+{% for blogPost in blogPosts %}
+    {{ blogPost.date|date('Y-m-d') }} &nbsp; <a href="{{ blogPost.url }}">{{ blogPost.title }}</a>
```
Comment (Ocramius) [inline source/archive.html]: Let's use `<date/>` around this

Comment (jwage) [reply source/archive.html]: 👍Changed

File: lib/Blog/BlogPostRepository.php [hunk 4]
```diff
@@ -32,6 +33,16 @@ public function find(SourceFile $sourceFile) : BlogPost
         );
     }
 
+    /**
+     * @return BlogPost[]
+     */
+    public function findPaginated(int $page = 1, int $perPage = 10) : array
```
Comment (SenseException) [inline lib/Blog/BlogPostRepository.php]: How should this behave on values `< 1`? Throwing an exception? Can we use `assert` here?

Comment (jwage) [reply lib/Blog/BlogPostRepository.php]: Hmm, good question. I am not sure. Do you think we should do something?

Comment (SenseException) [reply lib/Blog/BlogPostRepository.php]: `array_slice` allows negative values for offset and length, but that's an implementation detail, that is not known to the code invoking `findPaginated()`. Are negative numbers allowed by the repository method? If not, then we should check them.

Comment (jwage) [reply lib/Blog/BlogPostRepository.php]: I get what you are saying. Do you think it is worth it in such a simple application where we are the ones calling findPaginated in 2 places?

Comment (Ocramius) [reply lib/Blog/BlogPostRepository.php]: Yes

**Pull request merged**